keps/sig-node/3063-dynamic-resource-allocation/README.md

```diff
@@ -0,0 +1,1537 @@
+   // (TBD – a separate controller might also work).
+   UsedBy []metav1.OwnerReference
+}
+
+type ResourceClaimPhase string
+
+const (
+    // The claim is waiting for allocation by the plugin.
+    //
+    // If it uses delayed allocation, SuitableNodes must have
+    // been set, otherwise the plugin cannot proceed with
+    // the allocation.
+    ResourceClaimPending ResourceClaimPhase = “Pending”
+
+    // Set by the plugin once the resource has been successfully
+    // allocated. The scheduler waits for all resources used by
+    // a Pod to be in this phase.
+    ResourceClaimAllocated ResourceClaimPhase = “Allocated”
+
+    // It can happen that a resource got allocated for a Pod and
+    // then the Pod cannot be scheduled onto the nodes where the allocated
+    // resource is available. The scheduler detects this and
+    // then sets the “reallocate” phase to tell the plugin that it must
+    // free the resource. The plugin does that and resets
+    // the ResourceClaimPhase back to "Pending".
+    ResourceClaimReallocate ResourceClaimPhase = “Reallocate”
+)
+
+type PodSpec {
+   ...
+   Resources []PodResource
+   ...
+}
+
+type Container {
+   ...
+   // The entries are the names of resources in PodSpec.Resources
+   // that are used by the container.
+   PodResources []string
+   ...
+}
+
+type PodResource struct {
+   // A name under which this resource can be referenced by the containers.
+   Name string
+
+   // The resource is independent of the Pod and defined by
+   // a separate ResourceClaim in the same namespace as
+   // the Pod. Either this or Template must be set, but not both.
+   ResourceClaimName *string
+
+    // Will be used to create a stand-alone ResourceClaim to allocate the resource.
+    // The pod in which this PodResource is embedded will be the
+    // owner of the ResourceClaim, i.e. the ResourceClaim will be deleted together with the
+    // pod.  The name of the ResourceClaim will be `<pod name>-<resource name>` where
+    // `<resource name>` is the name PodResource.Name
+    // Pod validation will reject the pod if the concatenated name
+    // is not valid for a ResourceClaim (for example, too long).
+    //
+    // An existing ResourceClaim with that name that is not owned by the pod
+    // will *not* be used for the pod to avoid using an unrelated
+    // resource by mistake. Starting the pod is then blocked until
+    // the unrelated ResourceClaim is removed. If such a pre-created ResourceClaim is
+    // meant to be used by the pod, the ResourceClaim has to be updated with an
+    // owner reference to the pod once the pod exists. Normally
+    // this should not be necessary, but it may be useful when
+    // manually reconstructing a broken cluster.
+    //
+    // This field is read-only and no changes will be made by Kubernetes
+    // to the ResourceClaim after it has been created.
+   Template *ResourceClaimTemplate
```

Comment:
In PodResource, we need to specify either the "ResourceClaimTemplate" or the "ResourceClaim", right? They are mutually exclusive, right?

Reply:
Yes. ResourceClaimName has "Either this or Template must be set, but not both." as comment. Is that enough or shall I repeat that also for Template?

Reply:
I think it is good to add it to the comment of the definition of PODResource struct.

Reply:
Done.